packages/application-shell/src/components/application-entry-point/application-entry-point.tsx

```diff
@@ -3,14 +3,51 @@ import type { TProviderProps } from '@commercetools-frontend/application-shell-c
 import { Children, ReactNode } from 'react';
 import { Redirect, Route, Switch } from 'react-router-dom';
 import invariant from 'tiny-invariant';
+import { useIsAuthorized } from '@commercetools-frontend/permissions';
+import { PageUnauthorized } from '@commercetools-frontend/application-components';
+import { entryPointUriPathToPermissionKeys } from '../../utils/formatters';
 import RouteCatchAll from '../route-catch-all';
 
 type Props<AdditionalEnvironmentProperties extends {}> = {
   environment: TProviderProps<AdditionalEnvironmentProperties>['environment'];
+  disableRoutePermissionCheck?: boolean;
   render?: () => JSX.Element;
   children?: ReactNode;
 };
 
+const ApplicationRouteWithPermissionCheck = <
+  AdditionalEnvironmentProperties extends {}
+>(
+  props: Props<AdditionalEnvironmentProperties>
+) => {
+  const permissionKeys = entryPointUriPathToPermissionKeys(
+    props.environment.entryPointUriPath
+  );
+  // Require View permission to render the application.
+  const canView = useIsAuthorized({
+    demandedPermissions: [permissionKeys.View],
+  });
+
+  if (canView) {
+    return <>{Children.only<ReactNode>(props.children)}</>;
+  }
+  return <PageUnauthorized />;
+};
+
+const ApplicationRoute = <AdditionalEnvironmentProperties extends {}>(
+  props: Props<AdditionalEnvironmentProperties>
+) => {
+  if (props.disableRoutePermissionCheck) {
```

Comment:
FYI: We would use this flag internally to bail out of this behavior, as permissions work a bit differently in our default applications.